### PR TITLE
Implement `with_st` to match non-`Send` arguments.

### PR DESCRIPTION
**Pull request by Monnoroch**
See #80.

### CODE REVIEW

File: mockall_derive/src/expectation.rs [hunk 1]
```diff
@@ -192,6 +192,16 @@ impl<'a> Common<'a> {
                 self.common.withf(__mockall_f);
                 self
             }
+
+            /// Single-threaded version of [`withf`](#method.withf).
+            /// Can be used when the argument or return type isn't `Send`.
```
Comment (asomers) [inline mockall_derive/src/expectation.rs]: The return type is irrelevant for `withf_st`

File: mockall/tests/automock_nonsend.rs [hunk 2]
```diff
@@ -19,3 +19,14 @@ fn returning_st() {
     let x = Rc::new(42u32);
     assert_eq!(43, *mock.foo(x).as_ref());
 }
+
```
Comment (asomers) [inline mockall/tests/automock_nonsend.rs]: While you're here, could you also add tests for `returning_st` and `withf_st` on static methods?

**Comment by Monnoroch** [comment]
Thanks Alan. I'm just not quite sure what to add to the changelog, it seems currently it's per-version and I don't know how to add an entry without knowing the next version number / date. Can you suggest something?

**Comment by asomers** [comment]
I just started a new section in the CHANGELOG.  If you rebase you'll get it.

**Comment by Monnoroch** [comment]
All done!

**Comment by asomers** [comment]
Thanks for your contribution!

**Pull request merged**